src/components/gallery/imageGallery/ImageModal.svelte

```diff
@@ -0,0 +1,146 @@
+<script lang="ts">
+  import { createEventDispatcher, onDestroy, onMount } from 'svelte'
+  import { galleryStore } from '../../../stores'
+  import type { Gallery, Image } from '$lib/gallery'
+  import { deleteImageFromWNFS } from '$lib/gallery'
+
+  export let image: Image
+  export let isModalOpen: boolean = false
+  let previousImage: Image | undefined
+  let nextImage: Image | undefined
+  let showPreviousArrow: boolean
+  let showNextArrow: boolean
+  let gallery: Gallery
+
+  const dispatch = createEventDispatcher()
+
+  const unsubcribe = galleryStore.subscribe(newState => (gallery = newState))
+
+  /**
+   * Close the modal and clear the `image` and set `isModalOpen to false
+   */
+  const handleCloseModal: () => void = () => {
+    image = null
+    previousImage = null
+    nextImage = null
+    isModalOpen = false
+    dispatch('close')
+  }
+
+  /**
+   * Delete an image from the user's WNFS
+   */
+  const handleDeleteImage: () => Promise<void> = async () => {
+    await deleteImageFromWNFS(image.name)
+    handleCloseModal()
+  }
+
+  /**
+   * Set the previous and next images to be toggled to when the arrows are clicked
+   */
+  const setCarouselState = () => {
+    const imageList = image.private
+      ? gallery.privateImages
+      : gallery.publicImages
+    const currentIndex = imageList.findIndex(val => val.cid === image.cid)
+    previousImage =
+      imageList[currentIndex - 1] ?? imageList[imageList.length - 1]
+    nextImage = imageList[currentIndex + 1] ?? imageList[0]
+
+    showPreviousArrow = imageList.length > 1 && !!previousImage
+    showNextArrow = imageList.length > 1 && !!nextImage
+  }
+
+  /**
+   * Load the correct image when a user clicks the Next or Previous arrows
+   * @param direction
+   */
+  const handleNextOrPrevImage: (
+    direction: 'next' | 'prev'
+  ) => void = direction => {
+    image = direction === 'prev' ? previousImage : nextImage
+    setCarouselState()
+  }
+
+  onMount(() => {
+    setCarouselState()
+  })
+
+  // Unsubscribe from galleryStore updates
+  onDestroy(unsubcribe)
+</script>
+
+{#if !!image}
+  <!-- bind:checked can't be set to !!image, so we need to set it to a boolean(casting image as a boolean throws a svelte error, so we're using isModalOpen) -->
+  <input
+    type="checkbox"
+    id={`image-modal-${image.cid}`}
+    class="modal-toggle"
+    bind:checked={isModalOpen}
```

Comment:
Svelte doesn't allow for expressions here, so i had to pass in an `isModalOpen` boolean prop rather than just doing `!!image` or `Boolean(image)` ¯\_(ツ)_/¯